src/components/tabs/tab-title/tab-title.spec.js

```diff
@@ -0,0 +1,288 @@
+import React from 'react';
+import TestRenderer from 'react-test-renderer';
+import 'jest-styled-components';
+import { shallow } from 'enzyme';
+import TabTitle from './tab-title.component';
+import StyledTabTitle from './tab-title.style';
+import classicTheme from '../../../style/themes/classic';
+import { assertStyleMatch } from '../../../__spec_helper__/test-utils';
+import baseTheme from '../../../style/themes/base';
+
+function render(props) {
+  return shallow(
+    <TabTitle
+      title='Tab Title 1'
+      dataTabId='uniqueid1'
+      { ...props }
+    />
+  );
+}
+
+function renderStyles(props) {
+  return TestRenderer.create(
+    <StyledTabTitle
+      title='Tab Title 1'
+      dataTabId='uniqueid1'
+      { ...props }
+    />
+  );
+}
+
+describe('TabTitle', () => {
+  let wrapper;
+  it('renders as expected', () => {
+    wrapper = renderStyles();
+    expect(wrapper.toJSON()).toMatchSnapshot();
+  });
+
+  it('renders a title as its child with a text passed as a prop', () => {
+    wrapper = render();
+    expect(wrapper.children()).toHaveLength(1);
+    expect(wrapper.children().text()).toEqual('Tab Title 1');
+  });
+
+  it('contains custom className if passed as a prop', () => {
+    wrapper = render({ className: 'class' });
+    expect(wrapper.find('.class').exists()).toEqual(true);
+  });
+
+  describe('when clicked', () => {
+    it('triggers onClick function', () => {
+      const onClickFunction = jest.fn();
+      wrapper = render({ onClick: onClickFunction });
+      wrapper.simulate('click');
+      expect(onClickFunction).toHaveBeenCalled();
```

Comment:
this test nothing as you're triggering the click directly on the component with your mocked function

Reply:
unnecessary test removed